### ISSUE TITLE
Discourse forums do not follow "always open in" rules

### ISSUE THREAD

**Issue by jonhoo** [mozilla/multi-account-containers#1256]
- Is "Firefox will: Never remember history" in the Firefox Preferences/Options under "Privacy & Security > History" selected? No
- Are you using Firefox in a Private Window? No
- Can you see a grayed out but ticked Checkbox with the description "Enable Container Tabs" in the Firefox Preferences/Options under "Tabs"? Yes
- Multi-Account Containers Version: 6.0.0
- Operating System + Version: up-to-date Arch Linux (kernel 4.18.5)
- Firefox Version: 63.0b3
- Other installed Add-ons + Version + Enabled/Disabled-Status:
```
Amazon SMILE!	1.3	true	{1417a6e0-be73-4358-912c-5dce719b5791}
Better Ctrl-Tab	1.0	true	ctrltab@thesquareplanet.com
Firefox Multi-Account Containers	6.0.0	true	@testpilot-containers
HTTPS Everywhere	2018.8.22	true	https-everywhere-eff@eff.org
Stylus	1.4.20	true	{7a7a4a92-a2a0-41d1-9fd7-1e92480d612d}
Tampermonkey	4.7.5788	true	firefox@tampermonkey.net
uBlock Origin	1.16.20	true	uBlock0@raymondhill.net
```

### Actual behavior

Open https://internals.rust-lang.org/ or https://users.rust-lang.org/ in a container tab. Right-click and choose MAC -> Always Open in This Container (some of the time, the MAC entry doesn't even show up in the context menu; could be related). Open a new non-container tab, and browse to the same URL. Notice that you are *not* prompted to open the page in the previously used container.

### Expected behavior

The new tab should prompt you to open the tab in the chosen container.

### Notes

Could perhaps be related to #1253?

**Comment by lukastribus** [comment]
Yeah, that's what I mentioned in the notes of #1253, thanks for opening a new issue for this.

What ended up working around this problem for me was:
* remove the site from the container
* open the site in the default container
* click on the (I) botton (where you see the SSL certificate before the URL)
* use the `delete cookies and website data` (or similar, I don't have the english local here)

This appears to purge all data from this website in the default container. It is important that you do not open the website in the default container again, until you added it to your dedicated container. After adding it to the dedicated container, things work as expected.

So the problem seems to be that if some website data is saved in the default container while adding the website to different container, the prompt in the default context never appears, regardless of the `Always open in ...` setting.

The workaround is ok, but we do need the root cause to be fixed.

**Comment by beroal** [comment]
I have the same issue with https://support.mozilla.org/ , but lukastribus’ workaround does not work.

**Comment by teddy-gustiaux** [comment]
@lukastribus Thank you for the workaround! It has fixed the issue I was having with some Discourse forums as well as with other "regular" websites.

**Comment by SabreCat** [comment]
@lukastribus ahhhh thank you for the workaround! This issue was driving me up the wall, one site stubbornly ignoring all attempts to have it open in the desired container, but the cookie cleaning did the trick.

**Comment by grahamperrin** [comment]
Please, is this still an issue? 

Worth reviewing, given the many enhancements and fixes to Firefox over the past year or so.
